server/scripts/randomize-participants-status.ts

```diff
@@ -0,0 +1,401 @@
+const EMPLOYER_ID = 'ef963828-37db-44b4-ae5d-1e65b5a9b588';
+const START_DATE = '2020-01-01';
+
+// ids for participants whom already have added statuses
+const usedIds = [];
+
+// table data
+let partStatusArray = [];
+let cohortPartArray = [];
+let partPostHireArray = [];
+let returnOfServiceArray = [];
+
+let participantId = 0;
+let status = '';
+let site = 0;
+let isCurrent = false;
+let incrementedDate = '';
+let randomizeHiredSite: number;
+
+// increment table ids
+let psId = 1;
+let cId = 1;
+let pphId = 1;
+let rosId = 1;
+
+const partic_statuses = [
+  'offer_made',
+  'hired',
+  'interviewing',
+  'archived',
+  'rejected',
+  'peoi',
+  'engaged',
+  'unsuccessful_graduation',
+];
+
+const sites = [2222, 6666, 4444, 3333, 5151, 1111, 8888, 9999];
+const rejectAckStatuses = ['withdrawn', 'position filled', 'not qualified', 'not responsive'];
+const rejectedStatuses = ['hired by other', ...rejectAckStatuses];
+const archivedType = ['duplicate', 'employmentEnded'];
+const returnOfServiceStatuses = ['assigned-same-site', 'assigned-new-site'];
+const positionType = ['casual', 'permanent'];
+const employmentType = ['part-time', 'full-time'];
+
+const randomize = (n: number, reject?: boolean) => {
+  let value = 0;
+  // prevent duplicate participant ids from being used
+  // Sonarcloud complains about Math.random, ignoring as it's just being used for simple random numbers
+  if (reject) {
+    value = Math.floor(Math.random() * n); //NOSONAR
+    while (usedIds.includes(value)) {
+      value = Math.floor(Math.random() * n); //NOSONAR
+    }
+    usedIds.push(value);
+
+    return value;
+  }
+  return Math.floor(Math.random() * n); //NOSONAR
+};
+
+// create a random date based off the previous statuses date
+const randomizeDate = (d?: string) => {
+  const start = dayjs(d || START_DATE);
+  const daysToAdd = randomize(365);
+  const addedDate = start.add(daysToAdd, 'day').format('YYYY/MM/DD');
+  incrementedDate = addedDate;
+  return addedDate;
+};
+
+/***
+ * Generate random amount of participant statuses and all relating statuses
+ */
+export const randomizeParticipantStatus = async () => {
+  for (let i = 0; i < NUM_LOOPS; i++) {
+    participantId = randomize(499, true) + 1;
+    const randomIndex = randomize(partic_statuses.length);
+    status = partic_statuses[randomIndex];
+    site = sites[randomize(sites.length)];
+
+    randomizeDate();
+
+    generateFullStatusEntry(status);
+  }
+
+  await convertToCsv(psId, partStatusArray, 'participants_status.csv');
+  await convertToCsv(cId, cohortPartArray, 'cohort_participants.csv');
+  await convertToCsv(pphId, partPostHireArray, 'participant_post_hire_status.csv');
+  await convertToCsv(rosId, returnOfServiceArray, 'return_of_service_status.csv');
+};
+
+// loop through a status and add any related pre-statuses
+const generateFullStatusEntry = (status: string) => {
+  if (status === 'engaged') {
+    isCurrent = true;
+    partStatusArray.push(generateProspectingEntry());
```

Comment:
how about passing `current` as a param to `generateProspectingEntry?`
We'd better avoid updating or referencing global variables as much as possible.

Reply:
Yah, better idea!